pepper-apis/docs/specification/src/components/schemas.yml

```diff
@@ -1720,6 +1795,56 @@ Question.Picklist:
         type: array
         items:
           $ref: '#/Answer.Picklist'
+Question.Matrix:
+  allOf:
+    - $ref: '#/Question'
+    - type: object
+      required:
+        - selectMode
+        - matrixOptions
+        - matrixQuestions
+        - answers
+      properties:
+        questionType:
+          type: string
+          default: MATRIX
+          enum:
+            - MATRIX
+        selectMode:
+          type: string
+          description: |
+            * `MULTIPLE` - Allows user to select as many options as needed (checkbox)
+            * `SINGLE` - Only one selection is allowed inside group (radio button)
+          enum:
+            - MULTIPLE
+            - SINGLE
+        groups:
+
+          type: array
+          description: A list of groups used in the `matrixOptions` values. If there are no options which explicity request grouping, this will be an empty array.
+          minItems: 0
+          items:
+            type: object
+            required:
+              - identifier
+              - name
+            properties:
+              identifier:
+                type: string
```

Comment:
If the API allows ungrouped options (options with a `groupId` set to `null`) to be mixed with grouped options, let's either allow for a `null` identifier in the `MatrixGroup` type, or require a non-`null` `groupId` in the `MatrixOption` type. The current contract allows for a set of ungrouped options which are unable to have a prompt set for them.